packages/swing-store/src/bundleStore.js

```diff
@@ -0,0 +1,242 @@
+
+  const sqlDeleteBundle = db.prepare(`
+    DELETE FROM bundles
+    WHERE bundleID = ?
+  `);
+
+  function deleteBundle(bundleID) {
+    if (hasBundle(bundleID)) {
+      ensureTxn();
+      sqlDeleteBundle.run(bundleID);
+      noteExport(bundleArtifactName(bundleID), undefined);
+    }
+  }
+
+  /**
+   * Read a bundle and return it as a stream of data suitable for export to
+   * another store.
+   *
+   * Bundle artifact names should be strings of the form:
+   *   `bundle.${bundleID}`
+   *
+   * @param {string} name
+   *
+   * @yields {Uint8Array}
+   * @returns {AsyncIterable<Uint8Array>}
+   */
+  async function* exportBundle(name) {
+    typeof name === 'string' || Fail`artifact name must be a string`;
+    const parts = name.split('.');
+    const [type, bundleID] = parts;
+    // prettier-ignore
+    (parts.length === 2 && type === 'bundle') ||
+      Fail`expected artifact name of the form 'bundle.{bundleID}', saw ${q(name)}`;
+    const bundle = getBundle(bundleID);
+    bundle || Fail`bundle ${q(name)} not available`;
+    yield* Readable.from(Buffer.from(bundle.endoZipBase64));
+  }
+
+  const sqlGetBundleIDs = db.prepare(`
+    SELECT bundleID
+    FROM bundles
+  `);
+  sqlGetBundleIDs.pluck(true);
+
+  /**
+   * Obtain artifact metadata records for bundles contained in this store.
+   *
+   * @yields {[key: string, value: string]}
+   * @returns {Iterable<[key: string, value: string]>}
+   */
+  function* getExportRecords() {
+    for (const bundleID of sqlGetBundleIDs.iterate()) {
+      yield [bundleArtifactName(bundleID), bundleID];
+    }
+  }
+
+  async function* getArtifactNames() {
+    for (const bundleID of sqlGetBundleIDs.iterate()) {
+      yield bundleArtifactName(bundleID);
+    }
+  }
+
+  function computeSha512(bytes) {
+    const hash = createHash('sha512');
+    hash.update(bytes);
+    return hash.digest().toString('hex');
+  }
+
+  /**
+   * @param {string} name  Artifact name of the bundle
+   * @param {SwingStoreExporter} exporter  Whence to get the bits
+   * @param {string} bundleID  Bundle ID of the bundle
+   * @returns {Promise<void>}
+   */
+  async function importBundle(name, exporter, bundleID) {
+    const parts = name.split('.');
+    const [type, bundleIDkey] = parts;
+    // prettier-ignore
+    parts.length === 2 && type === 'bundle' ||
+      Fail`expected artifact name of the form 'bundle.{bundleID}', saw '${q(name)}'`;
+    bundleIDkey === bundleID ||
+      Fail`bundle artifact name ${name} doesn't match bundleID ${bundleID}`;
+    bundleID.startsWith('b1-') || Fail`invalid bundleID ${q(bundleID)}`;
+    const artifactChunks = exporter.getArtifact(name);
+    const inStream = Readable.from(artifactChunks);
+    const encodedBundle = await buffer(inStream);
+    const bundle = harden({
+      moduleFormat: 'endoZipBase64',
+      endoZipBase64Sha512: bundleID.substring(3),
+      endoZipBase64: encodedBundle.toString(),
+    });
+    await checkBundle(bundle, computeSha512, bundleID);
```

Comment:
let's add a comment, "assert the bundle contents match the ID", to make it clear where our security check is happening (in case the reader doesn't know quite what `checkBundle` actually checks)